commit 3b9de53dace31cc2a1b25d4cefe5e027d64e0432
Author: Ozan Çağlayan <ozan@pardus.org.tr>
Date:   Thu Feb 11 15:52:48 2010 +0200

    grabber: Support authenticated proxies
    
    Currently urlgrabber can't fetch files over an authenticated proxy
    because of not passing the authentication method. Passing HTTPAUTH_ANY
    handles all authentication methods like NTLM, Basic, Radius, etc.
    while keeping the behaviour intact if no authentication is required.
    
    Tested through an NTLM authenticated ISA proxy and a non-authenticated
    ISA proxy.

diff --git a/urlgrabber/grabber.py b/urlgrabber/grabber.py
index 90311de..74f06eb 100644
--- a/urlgrabber/grabber.py
+++ b/urlgrabber/grabber.py
@@ -1224,6 +1224,9 @@ class PyCurlFileObject():
                         continue
                     else:
                         if proxy == '_none_': proxy = ""
+                        # Don't fail if the remote end expects an authentication
+                        # Useful for authenticated proxy servers like ISA
+                        self.curl_obj.setopt(pycurl.PROXYAUTH, pycurl.HTTPAUTH_ANY)
                         self.curl_obj.setopt(pycurl.PROXY, proxy)
             
         # FIXME username/password/auth settings
